vllm/worker/hpu_enc_dec_model_runner.py

```diff
@@ -0,0 +1,606 @@
+                                                     is_prompt)
+                for i, b in enumerate(blocks)
+            ]
+        torch.hpu.synchronize()
+        profiler = None
+        if is_pt_profiler_run and self.is_driver_worker:
+            profiler = setup_profiler()
+            profiler.start()
+        for _ in range(times):
+            inputs = self.prepare_model_input(seqs)
+            self.execute_model(inputs, kv_caches, warmup_mode=True)
+            torch.hpu.synchronize()
+            if profiler:
+                profiler.step()
+        if profiler:
+            profiler.stop()
+        self.profiler.end()
+        gc.collect()
+
+    def create_dummy_seq_group_metadata(self,
+                                        group_id,
+                                        seq_len,
+                                        is_prompt,
+                                        lora_request=None,
+                                        temperature=0):
+        sampling_params = SamplingParams(temperature=0)
+        num_blocks = math.ceil(seq_len / self.block_size)
+        cross_block_table: Optional[List[int]] = None
+        encoder_dummy_data \
+            = self.input_registry.dummy_data_for_profiling(
+                self.model_config,
+                                        seq_len,
+                                        self.mm_registry,
+                                        is_encoder_data=True)
+        mm_counts = self.mm_registry.get_mm_limits_per_prompt(
+            self.model_config)
+        num_images = mm_counts["image"]
+        max_mm_tokens = self.mm_registry.get_max_multimodal_tokens(
+            self.model_config) * num_images
+        num_cross_blocks = math.ceil(max_mm_tokens / self.block_size)
+        seq_len = max(seq_len, 1)
+        if is_prompt:
+            input_len = seq_len
+            output_len = 0
+            block_tables = None
+            cross_block_table = None
+        else:
+            input_len = seq_len - 1
+            output_len = 1
+            block_tables = {group_id: [_PAD_BLOCK_ID] * num_blocks}
+            cross_block_table = [_PAD_BLOCK_ID] * num_cross_blocks
+        prompt_token_ids = [0] * input_len
+        output_token_ids = [1] * output_len
+        prompt_token_ids_array = array('l', prompt_token_ids)  # noqa: F821
+        seq_data = SequenceData(prompt_token_ids_array)
+        seq_data.output_token_ids = output_token_ids
+        return SequenceGroupMetadata(
+            request_id=str(group_id),
+            is_prompt=(output_len == 0),
+            seq_data={group_id: seq_data},
+            sampling_params=sampling_params,
+            block_tables=block_tables,
+            encoder_seq_data=encoder_dummy_data.seq_data,
+            multi_modal_data=encoder_dummy_data.multi_modal_data,
+            cross_block_table=cross_block_table)
+
+    def trim_attn_metadata(self, metadata: AttentionMetadata) -> object:
```

Comment:
do we need to define new function? can't we derive function from HPUModelRunnerBase?
It looks like a 1:1 copy

Reply:
No, we can't. We need to construct much more fields in the `TrimmedAttentionMetadata` than the base class. This also determines whether the graph will be recaptured in each inference step because all of its fields will be passed to the model execution as inputs. We should be very cautious in constructing them.